cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaSetOperator.java

```diff
@@ -49,4 +58,64 @@ public static boolean needsRollingUpdate(StatefulSetDiff diff) {
         }
         return false;
     }
+
+    private void revertVolumeChanges(StatefulSet current, StatefulSet desired) {
+
+        Container currentKafka =
+                current.getSpec().getTemplate().getSpec().getContainers().stream().filter(c -> c.getName().equals("kafka")).findFirst().get();
+        Container desiredKafka =
+                desired.getSpec().getTemplate().getSpec().getContainers().stream().filter(c -> c.getName().equals("kafka")).findFirst().get();
+
+        desiredKafka.setVolumeMounts(currentKafka.getVolumeMounts());
+
+        StatefulSetBuilder ssBuilder = new StatefulSetBuilder(desired);
+        StatefulSet updated = ssBuilder
+                .editSpec()
+                    .editTemplate()
+                        .editSpec()
+                            .editFirstContainer()
+                                .editMatchingEnv(e -> e.getName().equals(KafkaCluster.ENV_VAR_KAFKA_LOG_DIRS))
+                                    .withValue(desiredKafka.getVolumeMounts().stream()
+                                    .filter(vm -> vm.getMountPath().contains(AbstractModel.VOLUME_NAME))
+                                    .map(vm -> vm.getMountPath()).collect(Collectors.joining(",")))
```

Comment:
`.collect` on it's own line, so the reader can easily see the sequence of operations on the stream.